app/src/main/java/io/gnosis/safe/ui/settings/app/passcode/RepeatPasscodeFragment.kt

```diff
@@ -83,8 +84,14 @@ class RepeatPasscodeFragment : BaseViewBindingFragment<FragmentPasscodeBinding>(
                                 dismissCreatePasscodeFragment()
                             },
                             confirmCallback = { dialog ->
-                                viewModel.enableBiometry()
-                                viewModel.encryptPasscodeWithBiometricKey(viewAction.passcode)
+                                try {
+                                    //Some device still throw an IllegalStateException
+                                    // (due to no fingerprints enrolled) even though they return BIOMETRIC_SUCCESS
+                                    viewModel.encryptPasscodeWithBiometricKey(viewAction.passcode)
```

Comment:
encryptPasscodeWithBiometry and enableBiometry are always used together => the whole try catch block can and should be moved to a viewModel.  enableBiometry method could not only set the setting but also do the biometric key preparation work.